### DISCUSSION TITLE
Doubt with the accuracy functión

### DISCUSSION THREAD

**Discussion by chechoreyes** [mrdbourke/pytorch-deep-learning#789] [Q&A]
Video 88
Time: 13:39:04

Hi, i have a question:

Why in this part: 

![fit-multiclass-loop](https://github.com/mrdbourke/pytorch-deep-learning/assets/19411589/43bb63a6-1479-452a-ba7e-bb015bf42db2)

In the accuracy function, you compares `y_bloob_test` (or train) and `test_preds` (or train), but i don't undestand. First,`y_blob_test` is a tensor with the labels and `y_pred` is the result of `torch.softmax(y_logits, dim=1).argmax(dim=1)` what are the highest probability indices predicted, then, you compares two different things or PyTorch assumes the value of that index?

Thanks!

**Comment by sametucuncu** [comment]
Put these commands below after the test_preds = torch.softmax... statement and analyze the outcomes.
```
print(f"y_blob_test: {y_blob_test}")
print(f"test_preds: {test_preds}")

```

**Comment by mrdbourke** [comment]
Hi @chechoreyes , 

Yes, you're correct.

We compare the labels (e.g. `y_blob_test`) with the predictions (e.g. `test_preds`) and see where they are equal to each other and sum the outputs.

We then divide by the number of samples to get accuracy as a percentage.

As in the `accuracy_fn()`: 

```python
# Calculate accuracy (a classification metric)
def accuracy_fn(y_true, y_pred):
    """Calculates accuracy between truth labels and predictions.

    Args:
        y_true (torch.Tensor): Truth labels for predictions.
        y_pred (torch.Tensor): Predictions to be compared to predictions.

    Returns:
        [torch.float]: Accuracy value between y_true and y_pred, e.g. 78.45
    """
    correct = torch.eq(y_true, y_pred).sum().item()
    acc = (correct / len(y_pred)) * 100
    return acc
```

Perhaps you could try enter two example tensors in the function above and see what happens?
